src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -537,6 +538,16 @@ public override void ValidateId(DocumentsOperationContext context, Slice lowerId
                 return;
             }
 
+            if (type == DocumentChangeTypes.Delete &&
+                config.BucketMigrations.TryGetValue(bucket, out var migration) &&
+                migration.Status == MigrationStatus.OwnershipTransferred)
```

Comment:
hmmm do we need to check for `DocumentFlags.Artificial / DocumentFlags.FromResharding` instead?

Reply:
:) done